src/MemoryRouter.test.tsx

```diff
@@ -199,6 +200,89 @@ describe("MemoryRouter", () => {
       it("prefetch should do nothing", async () => {
         expect(await memoryRouter.prefetch()).toBeUndefined();
       });
+
+      it("when dynamic path registered will parse variables from slug", async () =>  {
+        memoryRouter.registerPaths(["/entity/[id]/attribute/[name]", "/[...slug]"]);
+
+        await memoryRouter.push("/entity/101/attribute/everything");
+        expect(memoryRouter).toMatchObject({
+          pathname: "/entity/101/attribute/everything",
```

Comment:
Note: look into supporting:
```suggestion
          pathname: "/entity/[id]/attribute/[name]",
```

Reply:
I'm not even 100% sure if my comment is correct ... if  you call `router.push(string)`, does it preserve the URL as-is for the `pathname`?  Or does it normalize the path?
What about a URL that's manually typed?  Does Next normalize the `pathname` there too?

Either way, I see you're still supporting the `router.push(UrlObject)` syntax, so this is fine to merge either way.

Reply:
I'm pretty sure that in order to push a route with the slugs uninterpolated, you have to provide a query dictionary along with the route. Otherwise there's no way that Next can determine what values the slugs should be replaced with. So `router.push(string)` should never have slugs in it.

Reply:
I was playing around with the feature in my own project, and I see what you're saying here now. I'm working on an update now.

Reply:
addressed in the latest push